In [1]:
#Nettoyage des données
import pandas as pd
import numpy as np

## Importation des données

In [46]:
#IMPORTATION DES DONNÉES

df_logement_anscien = pd.read_csv('dpe_data_complete_20251026_104213.csv', sep=';')
df_logement_neuf = pd.read_csv('dpe_data_complete_neuf_20251026_172218.csv', sep=';')

C:\Users\gopit\AppData\Local\Temp\ipykernel_28420\511196890.py:3: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_logement_anscien = pd.read_csv('dpe_data_complete_20251026_104213.csv', sep=';')


## Visualisation des deux datasets

In [52]:
df_logement_anscien.head(5)

conso_5_usages_ef  conso_5_usages_ef_energie_n1  \
0            27197.3                       26019.9   
1            43666.1                       24049.0   
2            23582.8                       23582.8   
3             8533.5                        8533.5   
4            52983.8                       52983.8   

             configuration_installation_chauffage_n1  \
0                   Installation de chauffage simple   
1                   Installation de chauffage simple   
2  Installation de chauffage avec en appoint un i...   
3                                                NaN   
4                   Installation de chauffage simple   

   conso_chauffage_installation_chauffage_n1  code_departement_ban  \
0                                    23676.5                  69.0   
1                                    24049.0                  69.0   
2                                        1.0                  69.0   
3                                        NaN                  69.0   
4                                        1.0                  69.0   

  code_insee_ban type_installation_chauffage_n1  \
0          69186      installation individuelle   
1          69161      installation individuelle   
2          69182      installation individuelle   
3          69182                            NaN   
4          69161      installation individuelle   

            type_generateur_n1_installation_n1  \
0  Chaudière fioul basse température 1991-2015   
1              Chaudière bois bûche avant 1978   
2               Cuisinière installé avant 1990   
3                                          NaN   
4           Chaudière fioul standard 1991-2015   

                    complement_adresse_logement  \
0  Ce bien ne fait pas partie d'une copropriété   
1  Ce bien ne fait pas partie d'une copropriété   
2  Ce bien ne fait pas partie d'une copropriété   
3  Ce bien ne fait pas partie d'une copropriété   
4                                           NaN   

                                 adresse_brut  ...  code_postal_ban  \
0                       532 route de Rochenin  ...            69790   
1                13, route de la Croix d'Ajou  ...            69790   
2                       290, chemin du vernay  ...            69790   
3  718 Route des Prôles \nLieu-dit Les Vignys  ...            69790   
4                 2101, route du Saint Rigaud  ...            69790   

   surface_habitable_logement etiquette_dpe     _score  code_postal_source  \
0                       133.6             E  44.682632               69790   
1                        94.0             G  44.682632               69790   
2                        50.1             G  44.682632               69790   
3                       166.2             C  44.682632               69790   
4                       134.2             G  44.682632               69790   

  annee_recherche      date_extraction            nom_rue_ban  \
0            2021  2025-10-26 09:27:34                    NaN   
1            2021  2025-10-26 09:27:34                    NaN   
2            2021  2025-10-26 09:27:34       Chemin du Vernay   
3            2021  2025-10-26 09:27:34                    NaN   
4            2021  2025-10-26 09:27:34  Route du Saint-Rigaud   

   type_installation_chauffage nombre_appartement  
0                          NaN                NaN  
1                          NaN                NaN  
2                          NaN                NaN  
3                          NaN                NaN  
4                          NaN                NaN  

[5 rows x 41 columns]

In [53]:
print(df_logement_anscien.info())
print(df_logement_anscien.shape)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1170868 entries, 0 to 1170867
Data columns (total 41 columns):
 #   Column                                      Non-Null Count    Dtype  
---  ------                                      --------------    -----  
 0   conso_5_usages_ef                           1170840 non-null  float64
 1   conso_5_usages_ef_energie_n1                1170840 non-null  float64
 2   configuration_installation_chauffage_n1     1139772 non-null  object 
 3   conso_chauffage_installation_chauffage_n1   1139748 non-null  float64
 4   code_departement_ban                        1169500 non-null  float64
 5   code_insee_ban                              1170868 non-null  object 
 6   type_installation_chauffage_n1              1139772 non-null  object 
 7   type_generateur_n1_installation_n1          1139772 non-null  object 
 8   complement_adresse_logement                 817532 non-null   object 
 9   adresse_brut                                1170852 non-n

In [54]:
df_logement_neuf.head(5)

emission_ges_5_usages_energie_n1  conso_5_usages_ef  \
0                              27.5             2669.8   
1                              27.5             2669.8   
2                              27.5             2669.8   
3                              27.5             2669.8   
4                            1251.7             5723.1   

   conso_5_usages_ef_energie_n1  code_departement_ban  \
0                        2669.8                  69.0   
1                        2669.8                  69.0   
2                        2669.8                  69.0   
3                        2669.8                  69.0   
4                        5349.0                  69.0   

   emission_ges_5_usages_par_m2  nombre_niveau_immeuble  \
0                           1.7                     1.0   
1                           1.7                     1.0   
2                           1.7                     1.0   
3                           1.7                     1.0   
4                          18.9                     NaN   

   surface_habitable_immeuble  conso_5_usages_ep  \
0                       116.6             6140.5   
1                       116.6             6140.5   
2                       116.6             6140.5   
3                       116.6             6140.5   
4                         NaN             6209.4   

                                        adresse_brut  code_region_ban  ...  \
0  20, Rue du Général de Gaulle - Lot 4 - Chemin ...             84.0  ...   
1  20, Rue du Général de Gaulle - Lot 4 - Chemin ...             84.0  ...   
2  20, Rue du Général de Gaulle - Lot 4 - Chemin ...             84.0  ...   
3  20, Rue du Général de Gaulle - Lot 4 - Chemin ...             84.0  ...   
4                                          de Verdun             84.0  ...   

   type_energie_n1         nom_commune_brut nombre_niveau_logement  \
0      Électricité  LE TREMBLAY SUR MAULDRE                    1.0   
1      Électricité  LE TREMBLAY SUR MAULDRE                    1.0   
2      Électricité  LE TREMBLAY SUR MAULDRE                    1.0   
3      Électricité  LE TREMBLAY SUR MAULDRE                    1.0   
4      Gaz naturel                   TARARE                    1.0   

   code_postal_ban  etiquette_dpe surface_habitable_logement     _score  \
0            69790              A                      116.6  55.134613   
1            69790              A                      116.6  55.134613   
2            69790              A                      116.6  55.134613   
3            69790              A                      116.6  55.134613   
4            69170              C                       67.8  34.385070   

  code_postal_source annee_recherche      date_extraction  
0              69790            2021  2025-10-26 17:03:18  
1              69790            2022  2025-10-26 17:03:20  
2              69790            2023  2025-10-26 17:03:22  
3              69790            2024  2025-10-26 17:03:24  
4              69170            2021  2025-10-26 17:03:27  

[5 rows x 41 columns]

In [55]:
print(df_logement_neuf.info())
print(df_logement_neuf.shape)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 160420 entries, 0 to 160419
Data columns (total 41 columns):
 #   Column                             Non-Null Count   Dtype  
---  ------                             --------------   -----  
 0   emission_ges_5_usages_energie_n1   160420 non-null  float64
 1   conso_5_usages_ef                  160420 non-null  float64
 2   conso_5_usages_ef_energie_n1       160420 non-null  float64
 3   code_departement_ban               160356 non-null  float64
 4   emission_ges_5_usages_par_m2       160420 non-null  float64
 5   nombre_niveau_immeuble             147028 non-null  float64
 6   surface_habitable_immeuble         146456 non-null  float64
 7   conso_5_usages_ep                  160420 non-null  float64
 8   adresse_brut                       160400 non-null  object 
 9   code_region_ban                    160356 non-null  float64
 10  cout_total_5_usages                160420 non-null  float64
 11  conso_5_usages_par_m2_ep           1604

## Valeurs manquantes

In [56]:
#Valeur manquante dans chaque colonne (en pourcentage) (si 0 ne pas afficher) (vm = valeur manquante)
vm_percentage = df_logement_anscien.isnull().mean() * 100
vm_percentage = vm_percentage[vm_percentage > 0]
print(vm_percentage)

conso_5_usages_ef                              0.002391
conso_5_usages_ef_energie_n1                   0.002391
configuration_installation_chauffage_n1        2.655807
conso_chauffage_installation_chauffage_n1      2.657857
code_departement_ban                           0.116836
type_installation_chauffage_n1                 2.655807
type_generateur_n1_installation_n1             2.655807
complement_adresse_logement                   30.177270
adresse_brut                                   0.001367
code_region_ban                                0.116836
cout_total_5_usages                            0.003416
zone_climatique                                0.122302
surface_chauffee_installation_chauffage_n1     2.655807
annee_construction                            42.449533
qualite_isolation_murs                         0.007174
type_generateur_chauffage_principal            2.655807
classe_altitude                                0.122302
cout_total_5_usages_energie_n1                 0

In [57]:
#Valeur manquante dans chaque colonne (en pourcentage) (si 0 ne pas afficher) (vm = valeur manquante)
vm_percentage = df_logement_neuf.isnull().mean() * 100
vm_percentage = vm_percentage[vm_percentage > 0]
print(vm_percentage)

code_departement_ban               0.039895
nombre_niveau_immeuble             8.348086
surface_habitable_immeuble         8.704650
adresse_brut                       0.012467
code_region_ban                    0.039895
nom_rue_ban                        4.655280
qualite_isolation_plancher_bas     0.087271
nombre_appartement                 5.101608
nom_commune_brut                   0.002493
nombre_niveau_logement            20.004987
surface_habitable_logement         1.007356
dtype: float64


In [58]:
#Liste des colonnes qui n'on pas de valeur manquante
no_vm_columns = df_logement_anscien.columns[df_logement_anscien.isnull().sum() == 0].tolist()
print("Colonnes sans valeurs manquantes :", no_vm_columns)

Colonnes sans valeurs manquantes : ['code_insee_ban', 'hauteur_sous_plafond', 'nom_commune_ban', '_geopoint', 'coordonnee_cartographique_x_ban', 'type_batiment', 'periode_construction', 'type_energie_principale_chauffage', 'date_reception_dpe', 'coordonnee_cartographique_y_ban', 'type_energie_n1', 'code_postal_ban', 'etiquette_dpe', '_score', 'code_postal_source', 'annee_recherche', 'date_extraction']


In [59]:
#Liste des colonnes qui n'on pas de valeur manquante
no_vm_columns = df_logement_anscien.columns[df_logement_neuf.isnull().sum() == 0].tolist()
print("Colonnes sans valeurs manquantes :", no_vm_columns)

Colonnes sans valeurs manquantes : ['conso_5_usages_ef', 'conso_5_usages_ef_energie_n1', 'configuration_installation_chauffage_n1', 'code_departement_ban', 'type_generateur_n1_installation_n1', 'code_region_ban', 'cout_total_5_usages', 'zone_climatique', 'hauteur_sous_plafond', 'surface_chauffee_installation_chauffage_n1', 'nom_commune_ban', 'annee_construction', '_geopoint', 'qualite_isolation_murs', 'type_generateur_chauffage_principal', 'type_batiment', 'classe_altitude', 'periode_construction', 'type_energie_principale_chauffage', 'cout_total_5_usages_energie_n1', 'adresse_ban', 'type_energie_n1', 'numero_etage_appartement', 'code_postal_ban', '_score', 'code_postal_source', 'date_extraction', 'nom_rue_ban', 'type_installation_chauffage', 'nombre_appartement']


### 1. Valeur manquantes de logement_anscien

1. type_installation_chauffage (multiple choix)

In [61]:
#Explication sur "type_installation_chauffage"
df_logement_anscien['type_installation_chauffage'].value_counts()

type_installation_chauffage
individuel                      571588
collectif                       395684
mixte (collectif-individuel)     26832
Name: count, dtype: int64

In [67]:
#Valeur manquantes de logement_anscien avec df_logement_anscien (pourcentage)
(df_logement_anscien['type_installation_chauffage'].isnull().sum()*100/len(df_logement_anscien)).item()

15.096834143558453

In [68]:
df_logement_anscien['type_installation_chauffage'].isnull().sum()

np.int64(176764)

In [70]:
1366741-176764

1189977

Je choisi de supprimé ligne qui on des valeur manquante de cette colonne

In [72]:
#Suppression les lignes avec des valeurs manquantes dans la colonne "type_installation_chauffage"
df_logement_anscien = df_logement_anscien.dropna(subset=['type_installation_chauffage'])
df_logement_anscien['type_installation_chauffage'].isnull().sum()


np.int64(0)

2. type_energie_principale_chauffage (multichoix)

In [74]:
#Exploration "type_energie_principale_chauffage"
df_logement_anscien['type_energie_principale_chauffage'].value_counts()

type_energie_principale_chauffage
Gaz naturel                                                     477192
Électricité                                                     335852
Réseau de Chauffage urbain                                      167420
Fioul domestique                                                  5924
Bois – Granulés (pellets) ou briquettes                           4268
Bois – Bûches                                                     1556
GPL                                                               1100
Propane                                                            536
Bois – Plaquettes d’industrie                                      116
Bois – Plaquettes forestières                                       80
Charbon                                                             44
Électricité d'origine renouvelable utilisée dans le bâtiment         8
Butane                                                               8
Name: count, dtype: int64

In [75]:
#Valeur manquantes de logement_anscien avec df_logement_anscien (pourcentage)
(df_logement_anscien['type_energie_principale_chauffage'].isnull().sum()*100/len(df_logement_anscien)).item()

0.0

In [78]:
#Regroupement des types d'énergie principale de chauffage
energie_regroupement = {
    'Gaz naturel': 'Gaz naturel',
    'Électricité': 'Électricité',
    'Réseau de Chauffage urbain': 'Réseau de chauffage urbain',
    'Fioul domestique': 'Fioul',
    'Bois – Granulés (pellets) ou briquettes': 'Bois et biomasse',
    'Bois – Bûches': 'Bois et biomasse',
    'Bois – Plaquettes d’industrie': 'Bois et biomasse',
    'Bois – Plaquettes forestières': 'Bois et biomasse',
    'GPL': 'Gaz (GPL/Propane/Butane)',
    'Propane': 'Gaz (GPL/Propane/Butane)',
    'Butane': 'Gaz (GPL/Propane/Butane)',
    'Charbon': 'Charbon',
    'Électricité d\'origine renouvelable utilisée dans le bâtiment': 'Électricité'
}

# Appliquer le mapping
df_logement_anscien['type_energie_principale_chauffage'] = df_logement_anscien['type_energie_principale_chauffage'].map(energie_regroupement)


In [79]:
#Exploration "type_energie_principale_chauffage"
df_logement_anscien['type_energie_principale_chauffage'].value_counts()

type_energie_principale_chauffage
Gaz naturel                   477192
Électricité                   335860
Réseau de chauffage urbain    167420
Bois et biomasse                6020
Fioul                           5924
Gaz (GPL/Propane/Butane)        1644
Charbon                           44
Name: count, dtype: int64

3. configuration_installation_chauffage_n1

In [80]:
#Explication sur "configuration_installation_chauffage_n1"
df_logement_anscien['configuration_installation_chauffage_n1'].value_counts()

configuration_installation_chauffage_n1
Installation de chauffage simple                                                                                                                         958424
Installation de chauffage collectif avec Base + appoint                                                                                                    7500
Installation de chauffage avec insert ou poêle bois en appoint                                                                                             2904
Convecteurs bi-jonction                                                                                                                                    1112
Installation de chauffage avec une chaudière ou une PAC en relève d’une chaudière bois                                                                      488
Installation de chauffage avec chaudière en relève de PAC                                                                                                   240


In [81]:
#Valeur manquantes de logement_anscien avec df_logement_anscien (pourcentage)
(df_logement_anscien['configuration_installation_chauffage_n1'].isnull().sum()*100/len(df_logement_anscien)).item()

2.315250718234712

In [82]:
df_logement_anscien['configuration_installation_chauffage_n1'].isnull().sum()

np.int64(23016)

Suppression de la colonne. Pour ne pas supprimer encore des lignes. Simplifier le dataframe, eviter overfitting sur des categorie compliqué (qualitative)

In [85]:
#Suppression de la colonnes. 
df_logement_anscien = df_logement_anscien.drop(columns=['configuration_installation_chauffage_n1'])


4. Complement_adresse_logement

In [86]:
#Explication "complement_adresse_logement "
df_logement_anscien['complement_adresse_logement'].value_counts()

complement_adresse_logement
Etage 1                                                                         23164
Etage 2                                                                         19564
Etage 3                                                                         14072
Etage RDC                                                                       12516
Etage : NC                                                                      11300
                                                                                ...  
413033 BAT:C715-01 - ENT:C715-01-01 - MOD:413033 - CodeMission:413033-E42871        4
413029 BAT:C715-01 - ENT:C715-01-01 - MOD:413029 - CodeMission:413029-E42871        4
412992 BAT:C715-01 - ENT:C715-01-01 - MOD:412992 - CodeMission:412992-E42871        4
413028 BAT:C715-01 - ENT:C715-01-01 - MOD:413028 - CodeMission:413028-E42871        4
Duplex partie Gauche du bâtiment                                                    4
Name: count, Length: 82970

In [87]:
(df_logement_anscien['complement_adresse_logement'].isnull().sum()*100/len(df_logement_anscien)).item()

22.00574587769489

Trop de valeur manquante pour un complement de l'adresse (22%). Il est preferable de supprimé la colonne. 

In [88]:
#Supprimener la colonne "complement_adresse_logement " 
df_logement_anscien = df_logement_anscien.drop(columns=['complement_adresse_logement']) 

5. periode_construction (0 valeurs manquantes) vs annee de construction (42% beaucoup trop important)

In [90]:
#Explication de "periode_construction"
df_logement_anscien['periode_construction'].value_counts()

periode_construction
1948-1974     353048
avant 1948    255600
2006-2012      88700
2013-2021      84408
1989-2000      75416
1975-1977      44188
1983-1988      34284
1978-1982      27916
2001-2005      25940
après 2021      4604
Name: count, dtype: int64

In [91]:
#Elplication de "Annee_construction"
df_logement_anscien['annee_construction'].value_counts()

annee_construction
1970.0    30852
1947.0    29512
1948.0    21524
1966.0    20196
1975.0    17884
          ...  
1855.0        4
1460.0        4
1812.0        4
1873.0        4
1886.0        4
Name: count, Length: 199, dtype: int64

42% des valeurs de la colonne "Annee_construction" sont manquantes. On va donc la supprimer puisqu'on a déjà la colonne "periode_construction" qui donne une idée de l'ancienneté du logement.

In [92]:

df_logement_anscien = df_logement_anscien.drop(columns=['annee_construction'])

6. type_generateur_chauffage_principal

In [93]:
#Explication "type_generateur_chauffage_principal"
df_logement_anscien['type_generateur_chauffage_principal'].value_counts()

type_generateur_chauffage_principal
Réseau de chaleur isolé                                      133984
Chaudière gaz à condensation après 2015                      127036
Chaudière gaz à condensation 2001-2015                       111524
Radiateur électrique à accumulation                           90028
Chaudière gaz standard 2001-2015                              88384
                                                              ...  
autre système à combustion fioul                                  4
poêle à bois bouilleur granulés installé à partir de 2012         4
Foyer fermé installé entre 1990 et 2004                           4
PAC eau glycolée/eau installée entre 2015 et 2016                 4
Chaudière gpl/propane/butane classique 1986-1990                  4
Name: count, Length: 123, dtype: int64

Trop de catégorie plus de 123, avec 2% de valeur manquante. Pas forcement plus d'apport que type d'energie principale. 

In [94]:
#Supprimer la colonne "type_generateur_chauffage_principal"
df_logement_anscien = df_logement_anscien.drop(columns=['type_generateur_chauffage_principal'])

7. nombre_appartement

In [95]:
#Explication de "nombre_appartement"
df_logement_anscien['nombre_appartement'].value_counts()

nombre_appartement
1.0      532052
20.0      29256
40.0      17144
30.0      13312
24.0      10504
          ...  
190.0         4
301.0         4
296.0         4
194.0         4
138.0         4
Name: count, Length: 265, dtype: int64

8. Surface_chauffe_installation_chauffage_n1 = surface_habitable_logement 

In [97]:
df_logement_anscien['surface_chauffee_installation_chauffage_n1'].head(5)

23    60.0
24    60.0
27    60.0
32    60.0
35    69.5
Name: surface_chauffee_installation_chauffage_n1, dtype: float64

In [98]:
df_logement_anscien['surface_habitable_logement'].head(5)

23    60.0
24    60.0
27    60.0
32    60.0
35    69.5
Name: surface_habitable_logement, dtype: float64

In [99]:
# Nombre de lignes avec valeurs manquantes dans les deux colonnes à la fois
missing_surface = df_logement_anscien[
    df_logement_anscien['surface_chauffee_installation_chauffage_n1'].isnull() &
    df_logement_anscien['surface_habitable_logement'].isnull()
]
print(missing_surface.shape[0])

536


In [100]:
#Supprimer ces lignes 
df_logement_anscien = df_logement_anscien.drop(missing_surface.index)

In [101]:
#Maintenant si il y a des valeurs manquantes dans "surface_chauffee_installation_chauffage_n1", on les remplit avec les valeurs de "surface_habitable_logement"
df_logement_anscien['surface_chauffee_installation_chauffage_n1'] = df_logement_anscien['surface_chauffee_installation_chauffage_n1'].fillna(
    df_logement_anscien['surface_habitable_logement'])

#Et l'inverse pour "surface_habitable_logement"
df_logement_anscien['surface_habitable_logement'] = df_logement_anscien['surface_habitable_logement'].fillna(
    df_logement_anscien['surface_chauffee_installation_chauffage_n1'])


In [104]:
(df_logement_anscien['surface_chauffee_installation_chauffage_n1'].isnull().sum()*100/len(df_logement_anscien)).item()

0.0

In [105]:
(df_logement_anscien['surface_habitable_logement'].isnull().sum()*100/len(df_logement_anscien)).item()

0.0

Puisque nous avons deja surface habitable et que les valeur sont les mêmes je suprime l'une des deux 'surface_chauffee_installation_chauffage_n1'

In [106]:
#supprimer la colonne 'surface_chauffee_installation_chauffage_n1'
df_logement_anscien = df_logement_anscien.drop(columns=['surface_chauffee_installation_chauffage_n1'])

9. Latitude et Longétude 

In [120]:
df_logement_anscien[['latitude', 'longitude']] = df_logement_anscien['_geopoint'].str.split(',', expand=True)

In [127]:
df_logement_anscien['longitude']

23         4.434598991483621
24         4.434598991483621
27         4.434598991483621
32         4.434598991483621
35         4.437585026879258
                 ...        
1170857    4.694136970169236
1170860    4.694722997951528
1170862    4.693923023910553
1170865    4.693978047177284
1170866    4.691196995331549
Name: longitude, Length: 993568, dtype: object

In [128]:
#Supprimer Latitude et longétude 
df_logement_anscien = df_logement_anscien.drop(columns=['_geopoint'])

Crée une colonne logement avec la valeur ancien

In [129]:
#Crée une colonne logement avec la valeur ancien. 
df_logement_anscien['logement'] = 'ancien'


In [130]:
df_logement_anscien['logement'] 

23         ancien
24         ancien
27         ancien
32         ancien
35         ancien
            ...  
1170857    ancien
1170860    ancien
1170862    ancien
1170865    ancien
1170866    ancien
Name: logement, Length: 993568, dtype: object

#### Recap 

Les colonnes avec le plus de valeur manquante on été traiter 

In [131]:
missing_percentage = df_logement_anscien.isnull().mean() * 100
missing_percentage = missing_percentage[missing_percentage > 0]
print(missing_percentage)

conso_5_usages_ef                            0.000403
conso_5_usages_ef_energie_n1                 0.000403
conso_chauffage_installation_chauffage_n1    2.262553
code_departement_ban                         0.098232
type_installation_chauffage_n1               2.262553
type_generateur_n1_installation_n1           2.262553
adresse_brut                                 0.001610
code_region_ban                              0.098232
cout_total_5_usages                          0.001610
zone_climatique                              0.066830
qualite_isolation_murs                       0.008454
classe_altitude                              0.066830
cout_total_5_usages_energie_n1               0.001208
adresse_ban                                  0.212567
numero_etage_appartement                     9.313504
nom_rue_ban                                  2.930046
nombre_appartement                           6.604883
dtype: float64


In [132]:
df_logement_anscien.shape

(993568, 39)

In [133]:
#liste des colonnes restantes toute la liste 
remaining_columns = df_logement_anscien.columns.tolist()
print(remaining_columns)

['conso_5_usages_ef', 'conso_5_usages_ef_energie_n1', 'conso_chauffage_installation_chauffage_n1', 'code_departement_ban', 'code_insee_ban', 'type_installation_chauffage_n1', 'type_generateur_n1_installation_n1', 'adresse_brut', 'code_region_ban', 'cout_total_5_usages', 'zone_climatique', 'hauteur_sous_plafond', 'nom_commune_ban', 'coordonnee_cartographique_x_ban', 'qualite_isolation_murs', 'type_batiment', 'classe_altitude', 'periode_construction', 'type_energie_principale_chauffage', 'date_reception_dpe', 'cout_total_5_usages_energie_n1', 'adresse_ban', 'coordonnee_cartographique_y_ban', 'type_energie_n1', 'numero_etage_appartement', 'code_postal_ban', 'surface_habitable_logement', 'etiquette_dpe', '_score', 'code_postal_source', 'annee_recherche', 'date_extraction', 'nom_rue_ban', 'type_installation_chauffage', 'nombre_appartement', 'energie_groupee', 'logement', 'latitude', 'longitude']


## Valeurs manquantes Logement Neuf 

In [142]:
df_logement_neuf.head(5)

emission_ges_5_usages_energie_n1  conso_5_usages_ef  \
0                              27.5             2669.8   
1                              27.5             2669.8   
2                              27.5             2669.8   
3                              27.5             2669.8   
4                            1251.7             5723.1   

   conso_5_usages_ef_energie_n1  code_departement_ban  \
0                        2669.8                  69.0   
1                        2669.8                  69.0   
2                        2669.8                  69.0   
3                        2669.8                  69.0   
4                        5349.0                  69.0   

   emission_ges_5_usages_par_m2  nombre_niveau_immeuble  \
0                           1.7                     1.0   
1                           1.7                     1.0   
2                           1.7                     1.0   
3                           1.7                     1.0   
4                          18.9                     NaN   

   surface_habitable_immeuble  conso_5_usages_ep  \
0                       116.6             6140.5   
1                       116.6             6140.5   
2                       116.6             6140.5   
3                       116.6             6140.5   
4                         NaN             6209.4   

                                        adresse_brut  code_region_ban  ...  \
0  20, Rue du Général de Gaulle - Lot 4 - Chemin ...             84.0  ...   
1  20, Rue du Général de Gaulle - Lot 4 - Chemin ...             84.0  ...   
2  20, Rue du Général de Gaulle - Lot 4 - Chemin ...             84.0  ...   
3  20, Rue du Général de Gaulle - Lot 4 - Chemin ...             84.0  ...   
4                                          de Verdun             84.0  ...   

   etiquette_dpe  surface_habitable_logement     _score  code_postal_source  \
0              A                       116.6  55.134613               69790   
1              A                       116.6  55.134613               69790   
2              A                       116.6  55.134613               69790   
3              A                       116.6  55.134613               69790   
4              C                        67.8  34.385070               69170   

   annee_recherche      date_extraction  anne_construction logement  \
0             2021  2025-10-26 17:03:18         après 2021     neuf   
1             2022  2025-10-26 17:03:20         après 2021     neuf   
2             2023  2025-10-26 17:03:22         après 2021     neuf   
3             2024  2025-10-26 17:03:24         après 2021     neuf   
4             2021  2025-10-26 17:03:27         après 2021     neuf   

             latitude          longitude  
0  46.216672021669055  4.434402972932283  
1  46.216672021669055  4.434402972932283  
2  46.216672021669055  4.434402972932283  
3  46.216672021669055  4.434402972932283  
4   45.89204299442378  4.436593038144964  

[5 rows x 44 columns]

In [144]:
df_logement_neuf.shape

(160420, 44)

1. type_energie_principale_chauffage

In [146]:
df_logement_neuf['type_energie_principale_chauffage'].value_counts()

type_energie_principale_chauffage
Gaz naturel                                76204
Électricité                                58008
Réseau de Chauffage urbain                 24112
Bois – Granulés (pellets) ou briquettes     1128
Bois – Plaquettes forestières                512
Bois – Plaquettes d’industrie                320
GPL                                          104
Bois – Bûches                                 32
Name: count, dtype: int64

In [147]:
#Regroupement des types d'énergie principale de chauffage
energie_regroupement = {
    'Gaz naturel': 'Gaz naturel',
    'Électricité': 'Électricité',
    'Réseau de Chauffage urbain': 'Réseau de chauffage urbain',
    'Bois – Granulés (pellets) ou briquettes': 'Bois et biomasse',
    'Bois – Bûches': 'Bois et biomasse',
    'Bois – Plaquettes d’industrie': 'Bois et biomasse',
    'Bois – Plaquettes forestières': 'Bois et biomasse',
    'GPL': 'Gaz (GPL/Propane/Butane)'
}

# Appliquer le mapping
df_logement_neuf['type_energie_principale_chauffage'] = df_logement_neuf['type_energie_principale_chauffage'].map(energie_regroupement)

In [148]:
df_logement_neuf['type_energie_principale_chauffage'].value_counts()

type_energie_principale_chauffage
Gaz naturel                   76204
Électricité                   58008
Réseau de chauffage urbain    24112
Bois et biomasse               1992
Gaz (GPL/Propane/Butane)        104
Name: count, dtype: int64

In [116]:
#Création colonne 'annee_construction' et 'logement'
df_logement_neuf['anne_construction'] = 'après 2021'
df_logement_neuf['logement'] = 'neuf'

2. Latitude et Longitude

In [134]:
df_logement_neuf[['latitude', 'longitude']] = df_logement_neuf['_geopoint'].str.split(',', expand=True)

In [135]:
df_logement_neuf['latitude']

0         46.216672021669055
1         46.216672021669055
2         46.216672021669055
3         46.216672021669055
4          45.89204299442378
                 ...        
160415     45.71486695832238
160416    45.724716977865064
160417    45.719397981744976
160418     45.72130801936334
160419    45.729212006772464
Name: latitude, Length: 160420, dtype: object

In [138]:
#Supprimer géopoint
df_logement_neuf = df_logement_neuf.drop(columns=['_geopoint'])

3. Surface

In [151]:
#Surface imeuble, surface_habitale, surface_chauffee
#surface_habitable_logement
#surface_habitable_immeuble

print(df_logement_neuf['surface_habitable_logement'].isnull().mean() * 100)
print(df_logement_neuf['surface_habitable_immeuble'].isnull().mean() * 100)




1.0073556913103103
8.704650292980926


In [152]:
missing_surface_neuf = df_logement_neuf[
    df_logement_neuf['surface_habitable_logement'].isnull() &
    df_logement_neuf['surface_habitable_immeuble'].isnull()
]
print(missing_surface_neuf.shape[0])

404


In [153]:
df_logement_neuf = df_logement_neuf.drop(missing_surface_neuf.index)

In [ ]:
print(df_logement_neuf['surface_habitable_logement'].isnull().mean() * 100)

0.7574242575742426


In [159]:
(df_logement_neuf['surface_habitable_logement'].isnull()).value_counts()

surface_habitable_logement
False    158804
True       1212
Name: count, dtype: int64

In [160]:
df_logement_neuf = df_logement_neuf.dropna(subset=['surface_habitable_logement'])

In [139]:
vm_percentage = df_logement_neuf.isnull().mean() * 100
vm_percentage = missing_percentage[missing_percentage > 0]
print(vm_percentage)

conso_5_usages_ef                            0.000403
conso_5_usages_ef_energie_n1                 0.000403
conso_chauffage_installation_chauffage_n1    2.262553
code_departement_ban                         0.098232
type_installation_chauffage_n1               2.262553
type_generateur_n1_installation_n1           2.262553
adresse_brut                                 0.001610
code_region_ban                              0.098232
cout_total_5_usages                          0.001610
zone_climatique                              0.066830
qualite_isolation_murs                       0.008454
classe_altitude                              0.066830
cout_total_5_usages_energie_n1               0.001208
adresse_ban                                  0.212567
numero_etage_appartement                     9.313504
nom_rue_ban                                  2.930046
nombre_appartement                           6.604883
dtype: float64


# Fusionner

In [163]:
#Fusionner les deux datasets 
colonne_communes = list(set(df_logement_anscien.columns) & set(df_logement_neuf.columns))
colonne_communes

['adresse_ban',
 '_score',
 'conso_5_usages_ef',
 'code_postal_ban',
 'nom_commune_ban',
 'nombre_appartement',
 'type_energie_principale_chauffage',
 'date_reception_dpe',
 'logement',
 'type_batiment',
 'latitude',
 'conso_5_usages_ef_energie_n1',
 'cout_total_5_usages',
 'coordonnee_cartographique_y_ban',
 'adresse_brut',
 'qualite_isolation_murs',
 'coordonnee_cartographique_x_ban',
 'surface_habitable_logement',
 'code_postal_source',
 'annee_recherche',
 'code_region_ban',
 'code_departement_ban',
 'nom_rue_ban',
 'cout_total_5_usages_energie_n1',
 'etiquette_dpe',
 'hauteur_sous_plafond',
 'longitude',
 'type_energie_n1',
 'date_extraction']

In [164]:
df_logement = pd.concat((df_logement_anscien, df_logement_neuf), join='inner')

# DATASET ENTIER

In [165]:
df_logement.shape

(1152372, 29)

In [169]:
df_logement_neuf.info()

<class 'pandas.core.frame.DataFrame'>
Index: 158804 entries, 0 to 160419
Data columns (total 44 columns):
 #   Column                             Non-Null Count   Dtype  
---  ------                             --------------   -----  
 0   emission_ges_5_usages_energie_n1   158804 non-null  float64
 1   conso_5_usages_ef                  158804 non-null  float64
 2   conso_5_usages_ef_energie_n1       158804 non-null  float64
 3   code_departement_ban               158740 non-null  float64
 4   emission_ges_5_usages_par_m2       158804 non-null  float64
 5   nombre_niveau_immeuble             145412 non-null  float64
 6   surface_habitable_immeuble         145244 non-null  float64
 7   conso_5_usages_ep                  158804 non-null  float64
 8   adresse_brut                       158800 non-null  object 
 9   code_region_ban                    158740 non-null  float64
 10  cout_total_5_usages                158804 non-null  float64
 11  conso_5_usages_par_m2_ep           158804 no

In [166]:
vm_percentage = df_logement.isnull().mean() * 100
vm_percentage = vm_percentage[vm_percentage > 0]
print(vm_percentage)

conso_5_usages_ef                 0.000347
conso_5_usages_ef_energie_n1      0.000347
code_departement_ban              0.090249
adresse_brut                      0.001736
code_region_ban                   0.090249
cout_total_5_usages               0.001388
qualite_isolation_murs            0.007289
cout_total_5_usages_energie_n1    0.001041
adresse_ban                       0.183274
nom_rue_ban                       3.162173
nombre_appartement                6.369471
dtype: float64


In [170]:
df_logement['conso_5_usages_ef'].isnull().sum()

np.int64(4)

In [171]:
df_logement = df_logement.dropna(subset=['conso_5_usages_ef'])

In [172]:
vm_percentage = df_logement.isnull().mean() * 100
vm_percentage = vm_percentage[vm_percentage > 0]
print(vm_percentage)

code_departement_ban              0.090249
adresse_brut                      0.001736
code_region_ban                   0.090249
cout_total_5_usages               0.001041
qualite_isolation_murs            0.007289
cout_total_5_usages_energie_n1    0.000694
adresse_ban                       0.183275
nom_rue_ban                       3.162184
nombre_appartement                6.369493
dtype: float64


In [175]:
df_logement['cout_total_5_usages'].isnull().sum()

np.int64(12)

In [176]:
df_logement = df_logement.dropna(subset=['cout_total_5_usages'])

In [177]:
vm_percentage = df_logement.isnull().mean() * 100
vm_percentage = vm_percentage[vm_percentage > 0]
print(vm_percentage)

code_departement_ban      0.090250
adresse_brut              0.001736
code_region_ban           0.090250
qualite_isolation_murs    0.007289
adresse_ban               0.183277
nom_rue_ban               3.162217
nombre_appartement        6.369559
dtype: float64


In [178]:
df_logement['qualite_isolation_murs'].isnull().sum()

np.int64(84)

In [179]:
df_logement.shape

(1152356, 29)

In [180]:
df_logement = df_logement.dropna(subset=['qualite_isolation_murs'])

In [181]:
df_logement.shape

(1152272, 29)

In [182]:
vm_percentage = df_logement.isnull().mean() * 100
vm_percentage = vm_percentage[vm_percentage > 0]
print(vm_percentage)

code_departement_ban    0.090256
adresse_brut            0.001736
code_region_ban         0.090256
adresse_ban             0.183290
nom_rue_ban             3.162448
nombre_appartement      6.370024
dtype: float64


In [183]:
df_logement['qualite_isolation_murs'].value_counts()

qualite_isolation_murs
insuffisante    554808
très bonne      248984
bonne           231980
moyenne         116500
Name: count, dtype: int64

In [185]:
df_logement["etiquette_dpe"].value_counts()

etiquette_dpe
C    472672
D    335772
E    179460
F     54748
B     52976
G     32488
A     24156
Name: count, dtype: int64

In [186]:
df_logement.to_csv('df_logement', sep=';', index=False)